server/services/offer.service.js

```diff
@@ -412,7 +412,7 @@ export const acceptOffer = async (offerToAccept) => {
 
   const vendor = await getUserById(offer.vendorId);
 
-  await activatePendingUserReferral({ userId: offer.userId, offerId: offer._id });
+  await activatePendingUserReferral({ userId: offer.userId, offer });
```

Comment:
I think it makes more sense to just pass `offer` only here since we can get the `userId` from offer.

```suggestion
  await activatePendingUserReferral({ offer });
```